example/flake.nix

```diff
@@ -51,7 +51,7 @@
               {
                 environment.PGWEB_DATABASE_URL = "postgres://$USER@${pgcfg.listen_addresses}:${builtins.toString pgcfg.port}/${dbName}";
                 command = pkgs.pgweb;
-                depends_on."pg1".condition = "process_healthy";
+                depends_on."pg1-init".condition = "process_completed_successfully";
```

Comment:
Why this change?

Reply:
Earlier we would use `postgres` CLI command to load the `initialScript` but as explained in https://github.com/juspay/services-flake/issues/63, it might not be the ideal way to load dumps/scripts. Switching to `psql` to do the same would require the `pg1-init` to run after `pg1`, hence the dependency changes. Although the naming of `pg1-init` should be changed as it is misleading.

Reply:
This would prove to be useful while resolving https://github.com/juspay/services-flake/issues/64 as `psql` works well with most of the dumps, but the `postgres` CLI command doesn't as seen in our case while trying to load these dumps: https://github.com/nammayatri/nammayatri/blob/94d8212c20043ad179e5b51b599deab4af8a4e8b/Backend/nix/arion-configuration.nix#L23-L38

Reply:
@srid I have added comments to move back to depending on `pg1` as we can use `pg_ctl` to temporarily start the postgres server while `pg1-init` is running.

Reply:
I have also refactored the service a bit, a lot of it can still be split into parts but this seemed enough based on what the PR is titled.

Reply:
LGTM